design/secretless-e2e-tests-dnd.md

```diff
@@ -0,0 +1,256 @@
+# Secretless automated end-to-end tests
+
+An R&amp;D Technical Design and Delivery Plan
+
+# References
+
+- Aha! Ticket: _Add link_
+- PRD Document: _Add link_
+- Feature Proposal: _Add link_
+
+# Overview
+
+The Secretless project needs to add automated end-to-end (e2e) tests for Conjur OSS. The current e2e automation system which relies on [the kubernetes-conjur-demo repo](https://github.com/conjurdemos/kubernetes-conjur-demo) is bloated and likely running redundant tests.
+
+Broadly speaking the optimisation needed are on the WHAT and the HOW of Secretless end-to-end tests.
+1. For the HOW, we focus on process and seek to reduce the complexity of orchestrating tests, to simplify maintenance and to make the testing pipeline developer-friendly.
+2. For the WHAT, we seek to optimise the scope of the tests to avoid doing more testing than is necessary, ideally the bulk of the tests will reside in tests at a lower level of granularity i.e. unit tests.
+
+The goal of this particular effort is to improve the process of creating and maintaining Secretless end-to-end tests. It is inspired by the immediate need to add a test case for Conjur OSS. Our current test cases assume success implicitly as running through some number of steps without encountering failure. WHAT we test requires some thought and improvement, however we deffer this to a later effort while keeping in mind that this addressing this technical debt will become simpler as a result of an improved process.
+
+The improvements in process are expected to allow developers on the Secretless project to focus efforts on writing integration test cases with the assurance that some robust other thing will handle
+1. dependency service lifecycle management
+2. running of the tests
+3. measuring and reporting the outcomes of the tests
+
+# Technical Objectives
+
+_At a high-level, what are the technical requirements that this design needs to satisfy?  How can it meet the acceptance criteria presented in the Feature Proposal?_
+
+- [ ] Tests run in a realistic environment with DAP/OSS as the source of truth for secret data
+- [ ] Tests run on developer machine ?
```

Comment:
I think if tests can be kicked off _from_ a developer machine, that would suffice